# Google_Play_Store Analysis and Visualization

## Problem Statement:

By using Pandas and numpy library analyse the google play Store dataset and 
Do the following task.

1.Import the library
2.Import the dataset
3.Remove the Duplicate values
4.Missing Value treatment
5.Make data  ready for implementing ML Model.


### Importing modules

In [1]:
import pandas as pd
import seaborn as sns
import numpy as np
import matplotlib.pyplot as plt


### Reading/Loading Dataset through Pandas

In [2]:
gp_store = pd.read_csv(r"C:\Users\Chani\Documents\Decodr\sample data\Googleplaystore (1).csv")

## description of Dataset

In [3]:
gp_store.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


#### which gives the column names, data type of elements present in column, range index and also gives the memory usage

In [4]:
gp_store.describe(include = "all")

App Category       Rating Reviews                Size    Installs  \
count    10841    10841  9367.000000   10841               10841       10841   
unique    9660       34          NaN    6002                 462          22   
top     ROBLOX   FAMILY          NaN       0  Varies with device  1,000,000+   
freq         9     1972          NaN     596                1695        1579   
mean       NaN      NaN     4.193338     NaN                 NaN         NaN   
std        NaN      NaN     0.537431     NaN                 NaN         NaN   
min        NaN      NaN     1.000000     NaN                 NaN         NaN   
25%        NaN      NaN     4.000000     NaN                 NaN         NaN   
50%        NaN      NaN     4.300000     NaN                 NaN         NaN   
75%        NaN      NaN     4.500000     NaN                 NaN         NaN   
max        NaN      NaN    19.000000     NaN                 NaN         NaN   

         Type  Price Content Rating Genres    Last Updated  \
count   10840  10841          10840  10841           10841   
unique      3     93              6    120            1378   
top      Free      0       Everyone  Tools  August 3, 2018   
freq    10039  10040           8714    842             326   
mean      NaN    NaN            NaN    NaN             NaN   
std       NaN    NaN            NaN    NaN             NaN   
min       NaN    NaN            NaN    NaN             NaN   
25%       NaN    NaN            NaN    NaN             NaN   
50%       NaN    NaN            NaN    NaN             NaN   
75%       NaN    NaN            NaN    NaN             NaN   
max       NaN    NaN            NaN    NaN             NaN   

               Current Ver Android Ver  
count                10833       10838  
unique                2832          33  
top     Varies with device  4.1 and up  
freq                  1459        2451  
mean                   NaN         NaN  
std                    NaN         NaN  
min                    NaN         NaN  
25%                    NaN         NaN  
50%                    NaN         NaN  
75%                    NaN         NaN  
max                    NaN         NaN

In [5]:
gp_store.describe(include= 'all').T

count unique                 top   freq      mean       std  \
App              10841   9660              ROBLOX      9       NaN       NaN   
Category         10841     34              FAMILY   1972       NaN       NaN   
Rating          9367.0    NaN                 NaN    NaN  4.193338  0.537431   
Reviews          10841   6002                   0    596       NaN       NaN   
Size             10841    462  Varies with device   1695       NaN       NaN   
Installs         10841     22          1,000,000+   1579       NaN       NaN   
Type             10840      3                Free  10039       NaN       NaN   
Price            10841     93                   0  10040       NaN       NaN   
Content Rating   10840      6            Everyone   8714       NaN       NaN   
Genres           10841    120               Tools    842       NaN       NaN   
Last Updated     10841   1378      August 3, 2018    326       NaN       NaN   
Current Ver      10833   2832  Varies with device   1459       NaN       NaN   
Android Ver      10838     33          4.1 and up   2451       NaN       NaN   

                min  25%  50%  75%   max  
App             NaN  NaN  NaN  NaN   NaN  
Category        NaN  NaN  NaN  NaN   NaN  
Rating          1.0  4.0  4.3  4.5  19.0  
Reviews         NaN  NaN  NaN  NaN   NaN  
Size            NaN  NaN  NaN  NaN   NaN  
Installs        NaN  NaN  NaN  NaN   NaN  
Type            NaN  NaN  NaN  NaN   NaN  
Price           NaN  NaN  NaN  NaN   NaN  
Content Rating  NaN  NaN  NaN  NaN   NaN  
Genres          NaN  NaN  NaN  NaN   NaN  
Last Updated    NaN  NaN  NaN  NaN   NaN  
Current Ver     NaN  NaN  NaN  NaN   NaN  
Android Ver     NaN  NaN  NaN  NaN   NaN

#### which gives the count of rows,mean,standard deviation,min and max value of each column and also gives 25% , 50% ,75 % of data

In [6]:
gp_store.shape

(10841, 13)

#### .shape gives the number of columns and number of rows in a Dataset

In [7]:
gp_store.dtypes

App                object
Category           object
Rating            float64
Reviews            object
Size               object
Installs           object
Type               object
Price              object
Content Rating     object
Genres             object
Last Updated       object
Current Ver        object
Android Ver        object
dtype: object

#### .dtype retuns the data type of elements in a column

In [8]:
gp_store.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

#### .head() prints the top 5 rows of a dataset

In [9]:
gp_store.tail()

App             Category  \
10836                               Sya9a Maroc - FR               FAMILY   
10837               Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                         Parkinson Exercices FR              MEDICAL   
10839                  The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840  iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating Reviews                Size     Installs  Type Price  \
10836     4.5      38                 53M       5,000+  Free     0   
10837     5.0       4                3.6M         100+  Free     0   
10838     NaN       3                9.5M       1,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   
10840     4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating             Genres      Last Updated         Current Ver  \
10836       Everyone          Education     July 25, 2017                1.48   
10837       Everyone          Education      July 6, 2018                 1.0   
10838       Everyone            Medical  January 20, 2017                 1.0   
10839     Mature 17+  Books & Reference  January 19, 2015  Varies with device   
10840       Everyone          Lifestyle     July 25, 2018  Varies with device   

              Android Ver  
10836          4.1 and up  
10837          4.1 and up  
10838          2.2 and up  
10839  Varies with device  
10840  Varies with device

#### .tail() Prints the bottom 5 rows of a dataset

In [10]:
gp_store.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

#### .columns prints the all column names of a datase

In [11]:
gp_store["App"].unique()

array(['Photo Editor & Candy Camera & Grid & ScrapBook',
       'Coloring book moana',
       'U Launcher Lite – FREE Live Cool Themes, Hide Apps', ...,
       'Parkinson Exercices FR', 'The SCP Foundation DB fr nn5n',
       'iHoroscope - 2018 Daily Horoscope & Astrology'], dtype=object)

#### .unique() prints all apps in a dataset

In [12]:
gp_store["App"].nunique()

9660

#### .nunique() gives the count of all apps in a dataset

In [13]:
gp_store["Category"].unique()

array(['ART_AND_DESIGN', 'AUTO_AND_VEHICLES', 'BEAUTY',
       'BOOKS_AND_REFERENCE', 'BUSINESS', 'COMICS', 'COMMUNICATION',
       'DATING', 'EDUCATION', 'ENTERTAINMENT', 'EVENTS', 'FINANCE',
       'FOOD_AND_DRINK', 'HEALTH_AND_FITNESS', 'HOUSE_AND_HOME',
       'LIBRARIES_AND_DEMO', 'LIFESTYLE', 'GAME', 'FAMILY', 'MEDICAL',
       'SOCIAL', 'SHOPPING', 'PHOTOGRAPHY', 'SPORTS', 'TRAVEL_AND_LOCAL',
       'TOOLS', 'PERSONALIZATION', 'PRODUCTIVITY', 'PARENTING', 'WEATHER',
       'VIDEO_PLAYERS', 'NEWS_AND_MAGAZINES', 'MAPS_AND_NAVIGATION',
       '1.9'], dtype=object)

In [14]:
gp_store["Category"].nunique()

34

In [15]:
len(gp_store["Category"].unique())

34

In [16]:
gp_store["Content Rating"].unique()

array(['Everyone', 'Teen', 'Everyone 10+', 'Mature 17+',
       'Adults only 18+', 'Unrated', nan], dtype=object)

In [17]:
gp_store["Content Rating"].nunique()

6

In [18]:
gp_store["Genres"].unique()

array(['Art & Design', 'Art & Design;Pretend Play',
       'Art & Design;Creativity', 'Art & Design;Action & Adventure',
       'Auto & Vehicles', 'Beauty', 'Books & Reference', 'Business',
       'Comics', 'Comics;Creativity', 'Communication', 'Dating',
       'Education;Education', 'Education', 'Education;Creativity',
       'Education;Music & Video', 'Education;Action & Adventure',
       'Education;Pretend Play', 'Education;Brain Games', 'Entertainment',
       'Entertainment;Music & Video', 'Entertainment;Brain Games',
       'Entertainment;Creativity', 'Events', 'Finance', 'Food & Drink',
       'Health & Fitness', 'House & Home', 'Libraries & Demo',
       'Lifestyle', 'Lifestyle;Pretend Play',
       'Adventure;Action & Adventure', 'Arcade', 'Casual', 'Card',
       'Casual;Pretend Play', 'Action', 'Strategy', 'Puzzle', 'Sports',
       'Music', 'Word', 'Racing', 'Casual;Creativity',
       'Casual;Action & Adventure', 'Simulation', 'Adventure', 'Board',
       'Trivia', 'Role 

In [19]:
gp_store["Genres"].nunique()

120

In [20]:
gp_store["Rating"].unique()

array([ 4.1,  3.9,  4.7,  4.5,  4.3,  4.4,  3.8,  4.2,  4.6,  3.2,  4. ,
        nan,  4.8,  4.9,  3.6,  3.7,  3.3,  3.4,  3.5,  3.1,  5. ,  2.6,
        3. ,  1.9,  2.5,  2.8,  2.7,  1. ,  2.9,  2.3,  2.2,  1.7,  2. ,
        1.8,  2.4,  1.6,  2.1,  1.4,  1.5,  1.2, 19. ])

In [21]:
gp_store["Rating"].nunique()

40

In [22]:
gp_store["Type"].unique()

array(['Free', 'Paid', nan, '0'], dtype=object)

In [23]:
gp_store["Type"].nunique()

3

In [24]:
gp_store["Last Updated"].unique()

array(['January 7, 2018', 'January 15, 2018', 'August 1, 2018', ...,
       'January 20, 2014', 'February 16, 2014', 'March 23, 2014'],
      dtype=object)

In [25]:
gp_store["Last Updated"].nunique()

1378

In [26]:
gp_store["Current Ver"].unique()

array(['1.0.0', '2.0.0', '1.2.4', ..., '1.0.612928', '0.3.4', '2.0.148.0'],
      dtype=object)

In [27]:
gp_store["Current Ver"].nunique()

2832

In [28]:
gp_store["Android Ver"].unique()

array(['4.0.3 and up', '4.2 and up', '4.4 and up', '2.3 and up',
       '3.0 and up', '4.1 and up', '4.0 and up', '2.3.3 and up',
       'Varies with device', '2.2 and up', '5.0 and up', '6.0 and up',
       '1.6 and up', '1.5 and up', '2.1 and up', '7.0 and up',
       '5.1 and up', '4.3 and up', '4.0.3 - 7.1.1', '2.0 and up',
       '3.2 and up', '4.4W and up', '7.1 and up', '7.0 - 7.1.1',
       '8.0 and up', '5.0 - 8.0', '3.1 and up', '2.0.1 and up',
       '4.1 - 7.1.1', nan, '5.0 - 6.0', '1.0 and up', '2.2 - 7.1.1',
       '5.0 - 7.1.1'], dtype=object)

In [29]:
gp_store["Android Ver"].nunique()

33

## Finding the null values

In [30]:
gp_store.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

#### .isnull() checks the column consits of null values or not
#### .sum() counts and gives how many null values present a column

### percentage of null values in a particular column

In [31]:
(gp_store.isnull().sum()/gp_store.shape[0])*100

App                0.000000
Category           0.000000
Rating            13.596532
Reviews            0.000000
Size               0.000000
Installs           0.000000
Type               0.009224
Price              0.000000
Content Rating     0.009224
Genres             0.000000
Last Updated       0.000000
Current Ver        0.073794
Android Ver        0.027673
dtype: float64

In [32]:
data1 = gp_store.head()
data2 = gp_store.tail()
concat_data = pd.concat([data1,data2],axis=0,ignore_index=True)
concat_data

App             Category  \
0     Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                Coloring book moana       ART_AND_DESIGN   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                              Sketch - Draw & Paint       ART_AND_DESIGN   
4              Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
5                                   Sya9a Maroc - FR               FAMILY   
6                   Fr. Mike Schmitz Audio Teachings               FAMILY   
7                             Parkinson Exercices FR              MEDICAL   
8                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
9      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

   Rating Reviews                Size     Installs  Type Price Content Rating  \
0     4.1     159                 19M      10,000+  Free     0       Everyone   
1     3.9     967                 14M     500,000+  Free     0       Everyone   
2     4.7   87510                8.7M   5,000,000+  Free     0       Everyone   
3     4.5  215644                 25M  50,000,000+  Free     0           Teen   
4     4.3     967                2.8M     100,000+  Free     0       Everyone   
5     4.5      38                 53M       5,000+  Free     0       Everyone   
6     5.0       4                3.6M         100+  Free     0       Everyone   
7     NaN       3                9.5M       1,000+  Free     0       Everyone   
8     4.5     114  Varies with device       1,000+  Free     0     Mature 17+   
9     4.5  398307                 19M  10,000,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   
5                  Education     July 25, 2017                1.48   
6                  Education      July 6, 2018                 1.0   
7                    Medical  January 20, 2017                 1.0   
8          Books & Reference  January 19, 2015  Varies with device   
9                  Lifestyle     July 25, 2018  Varies with device   

          Android Ver  
0        4.0.3 and up  
1        4.0.3 and up  
2        4.0.3 and up  
3          4.2 and up  
4          4.4 and up  
5          4.1 and up  
6          4.1 and up  
7          2.2 and up  
8  Varies with device  
9  Varies with device

## Duplicates

In [33]:
gp_store.duplicated().sum()

483

In [34]:
gp_store.App.duplicated().sum()

1181

In [35]:
gp_store[gp_store.App.duplicated()]

App       Category  Rating Reviews  \
229          Quick PDF Scanner + OCR FREE       BUSINESS     4.2   80805   
236                                   Box       BUSINESS     4.2  159872   
239                    Google My Business       BUSINESS     4.4   70991   
256                   ZOOM Cloud Meetings       BUSINESS     4.4   31614   
261             join.me - Simple Meetings       BUSINESS     4.0    6989   
...                                   ...            ...     ...     ...   
10715                  FarmersOnly Dating         DATING     3.0    1145   
10720  Firefox Focus: The privacy browser  COMMUNICATION     4.4   36981   
10730                         FP Notebook        MEDICAL     4.5     410   
10753      Slickdeals: Coupons & Shopping       SHOPPING     4.5   33599   
10768                                AAFP        MEDICAL     3.8      63   

                     Size     Installs  Type Price Content Rating  \
229    Varies with device   5,000,000+  Free     0       Everyone   
236    Varies with device  10,000,000+  Free     0       Everyone   
239    Varies with device   5,000,000+  Free     0       Everyone   
256                   37M  10,000,000+  Free     0       Everyone   
261    Varies with device   1,000,000+  Free     0       Everyone   
...                   ...          ...   ...   ...            ...   
10715                1.4M     100,000+  Free     0     Mature 17+   
10720                4.0M   1,000,000+  Free     0       Everyone   
10730                 60M      50,000+  Free     0       Everyone   
10753                 12M   1,000,000+  Free     0       Everyone   
10768                 24M      10,000+  Free     0       Everyone   

              Genres       Last Updated         Current Ver  \
229         Business  February 26, 2018  Varies with device   
236         Business      July 31, 2018  Varies with device   
239         Business      July 24, 2018    2.19.0.204537701   
256         Business      July 20, 2018      4.1.28165.0716   
261         Business      July 16, 2018           4.3.0.508   
...              ...                ...                 ...   
10715         Dating  February 25, 2016                 2.2   
10720  Communication       July 6, 2018                 5.2   
10730        Medical     March 24, 2018           2.1.0.372   
10753       Shopping      July 30, 2018                 3.9   
10768        Medical      June 22, 2018               2.3.1   

              Android Ver  
229          4.0.3 and up  
236    Varies with device  
239            4.4 and up  
256            4.0 and up  
261            4.4 and up  
...                   ...  
10715          4.0 and up  
10720          5.0 and up  
10730          4.4 and up  
10753          4.4 and up  
10768          5.0 and up  

[1181 rows x 13 columns]

#### printed the all duplicates present in App column

In [36]:
gp_store[gp_store.App == "AAFP"]

App Category  Rating Reviews Size Installs  Type Price Content Rating  \
2515   AAFP  MEDICAL     3.8      63  24M  10,000+  Free     0       Everyone   
10768  AAFP  MEDICAL     3.8      63  24M  10,000+  Free     0       Everyone   

        Genres   Last Updated Current Ver Android Ver  
2515   Medical  June 22, 2018       2.3.1  5.0 and up  
10768  Medical  June 22, 2018       2.3.1  5.0 and up

In [37]:
gp_store[gp_store.App == "FP Notebook"]

App Category  Rating Reviews Size Installs  Type Price  \
2379   FP Notebook  MEDICAL     4.5     408  60M  50,000+  Free     0   
10730  FP Notebook  MEDICAL     4.5     410  60M  50,000+  Free     0   

      Content Rating   Genres    Last Updated Current Ver Android Ver  
2379        Everyone  Medical  March 24, 2018   2.1.0.372  4.4 and up  
10730       Everyone  Medical  March 24, 2018   2.1.0.372  4.4 and up

In [38]:
gp_store[gp_store.App == "Box"]

App  Category  Rating Reviews                Size     Installs  Type  \
204  Box  BUSINESS     4.2  159872  Varies with device  10,000,000+  Free   
236  Box  BUSINESS     4.2  159872  Varies with device  10,000,000+  Free   
265  Box  BUSINESS     4.2  159872  Varies with device  10,000,000+  Free   

    Price Content Rating    Genres   Last Updated         Current Ver  \
204     0       Everyone  Business  July 31, 2018  Varies with device   
236     0       Everyone  Business  July 31, 2018  Varies with device   
265     0       Everyone  Business  July 31, 2018  Varies with device   

            Android Ver  
204  Varies with device  
236  Varies with device  
265  Varies with device

### viewed the duplicates 

##  Analysis

### App

In [39]:
gp_store["App"].unique()

array(['Photo Editor & Candy Camera & Grid & ScrapBook',
       'Coloring book moana',
       'U Launcher Lite – FREE Live Cool Themes, Hide Apps', ...,
       'Parkinson Exercices FR', 'The SCP Foundation DB fr nn5n',
       'iHoroscope - 2018 Daily Horoscope & Astrology'], dtype=object)

In [40]:
gp_store["App"].nunique()

9660

In [41]:
gp_store["App"].isnull().sum()

0

#### App column have 9660 apps and  no null values and we can't futher any analysis

### Category

In [42]:
gp_store["Category"].isnull().sum()

0

In [43]:
gp_store["Category"].nunique()

34

#### Category column have 34 categories and had no null values 

In [44]:
gp_store["Category"].unique()

array(['ART_AND_DESIGN', 'AUTO_AND_VEHICLES', 'BEAUTY',
       'BOOKS_AND_REFERENCE', 'BUSINESS', 'COMICS', 'COMMUNICATION',
       'DATING', 'EDUCATION', 'ENTERTAINMENT', 'EVENTS', 'FINANCE',
       'FOOD_AND_DRINK', 'HEALTH_AND_FITNESS', 'HOUSE_AND_HOME',
       'LIBRARIES_AND_DEMO', 'LIFESTYLE', 'GAME', 'FAMILY', 'MEDICAL',
       'SOCIAL', 'SHOPPING', 'PHOTOGRAPHY', 'SPORTS', 'TRAVEL_AND_LOCAL',
       'TOOLS', 'PERSONALIZATION', 'PRODUCTIVITY', 'PARENTING', 'WEATHER',
       'VIDEO_PLAYERS', 'NEWS_AND_MAGAZINES', 'MAPS_AND_NAVIGATION',
       '1.9'], dtype=object)

#### In category column has "1.9" is a wrong value and check it

In [45]:
gp_store[gp_store["Category"] == '1.9']

App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs Type     Price Content Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  
10472       1.0.19  4.0 and up         NaN

### we can create NaN category using the shift function

In [46]:
gp_store.loc[10472] = gp_store.loc[10472].shift()
gp_store['App'].loc[10472] = gp_store['Category'].loc[10472]
gp_store['Category'].loc[10472] = np.nan
gp_store.loc[10472]

App               Life Made WI-Fi Touchscreen Photo Frame
Category                                              NaN
Rating                                                1.9
Reviews                                              19.0
Size                                                 3.0M
Installs                                           1,000+
Type                                                 Free
Price                                                   0
Content Rating                                   Everyone
Genres                                                NaN
Last Updated                            February 11, 2018
Current Ver                                        1.0.19
Android Ver                                    4.0 and up
Name: 10472, dtype: object

In [47]:
gp_store[gp_store["Category"] == '1.9']

Empty DataFrame
Columns: [App, Category, Rating, Reviews, Size, Installs, Type, Price, Content Rating, Genres, Last Updated, Current Ver, Android Ver]
Index: []

In [48]:
gp_store["Category"].isnull().sum()

1

In [49]:
gp_store["Category"].replace(np.nan,"Other",inplace = True)

In [50]:
gp_store["Category"].isnull().sum()

0

### Rating

In [51]:
gp_store["Rating"].unique()

array([4.1, 3.9, 4.7, 4.5, 4.3, 4.4, 3.8, 4.2, 4.6, 3.2, 4.0, nan, 4.8,
       4.9, 3.6, 3.7, 3.3, 3.4, 3.5, 3.1, 5.0, 2.6, 3.0, 1.9, 2.5, 2.8,
       2.7, 1.0, 2.9, 2.3, 2.2, 1.7, 2.0, 1.8, 2.4, 1.6, 2.1, 1.4, 1.5,
       1.2, '1.9'], dtype=object)

In [52]:
gp_store["Rating"].isnull().sum()

1474

In [53]:
gp_store["Rating"].isnull().value_counts()

False    9367
True     1474
Name: Rating, dtype: int64

In [54]:
gp_store["Rating"].isna().value_counts()

False    9367
True     1474
Name: Rating, dtype: int64

#### Rating column have 40 types of ratings and had 1474 null values in rating column

#### Rating is  contious data and has no outliers , so we repalce the nan values with the mean

In [55]:
gp_store['Rating'] = pd.to_numeric(gp_store['Rating'], errors='coerce')

In [56]:
gp_store["Rating"].mean()

4.1915127575531175

In [57]:
gp_store["Rating"].replace(np.nan,gp_store["Rating"].mean(),inplace = True)

In [58]:
gp_store["Rating"].notnull().sum()

10841

In [59]:
gp_store["Rating"].isnull().sum()

0

In [60]:
gp_store["Rating"].unique()

array([4.1       , 3.9       , 4.7       , 4.5       , 4.3       ,
       4.4       , 3.8       , 4.2       , 4.6       , 3.2       ,
       4.        , 4.19151276, 4.8       , 4.9       , 3.6       ,
       3.7       , 3.3       , 3.4       , 3.5       , 3.1       ,
       5.        , 2.6       , 3.        , 1.9       , 2.5       ,
       2.8       , 2.7       , 1.        , 2.9       , 2.3       ,
       2.2       , 1.7       , 2.        , 1.8       , 2.4       ,
       1.6       , 2.1       , 1.4       , 1.5       , 1.2       ])

In [61]:
gp_store.count()

App               10841
Category          10841
Rating            10841
Reviews           10841
Size              10841
Installs          10841
Type              10840
Price             10841
Content Rating    10841
Genres            10840
Last Updated      10841
Current Ver       10833
Android Ver       10839
dtype: int64

### Reviews

In [62]:
gp_store["Reviews"].unique()

array(['159', '967', '87510', ..., '603', '1195', '398307'], dtype=object)

In [63]:
gp_store["Reviews"].nunique()

6002

In [64]:
gp_store[gp_store["Reviews"] =="3.0M"]

Empty DataFrame
Columns: [App, Category, Rating, Reviews, Size, Installs, Type, Price, Content Rating, Genres, Last Updated, Current Ver, Android Ver]
Index: []

In [65]:
gp_store['Reviews'] = gp_store.Reviews.replace("0.0",0)
gp_store['Reviews'] = gp_store.Reviews.replace("3.0M",3000000.0)
gp_store['Reviews'] = gp_store['Reviews'].astype(float)
gp_store['Reviews'].dtype

dtype('float64')

In [66]:
gp_store["Reviews"].max()

78158306.0

In [67]:
gp_store["Reviews"].min()

0.0

### Size

In [68]:
gp_store["Size"].isnull().sum()

0

#### size column has no null values and size may vary from one App to another App

In [69]:
gp_store[gp_store["Size"] == 'Varies with device']

App             Category  \
37                                    Floor Plan Creator       ART_AND_DESIGN   
42                            Textgram - write on photos       ART_AND_DESIGN   
52                         Used Cars and Trucks for Sale    AUTO_AND_VEHICLES   
67                                    Ulysse Speedometer    AUTO_AND_VEHICLES   
68                                                REPUVE    AUTO_AND_VEHICLES   
...                                                  ...                  ...   
10713  My Earthquake Alerts - US & Worldwide Earthquakes              WEATHER   
10725                                          Posta App  MAPS_AND_NAVIGATION   
10765                    Chat For Strangers - Video Chat               SOCIAL   
10826          Frim: get new friends on local chat rooms               SOCIAL   
10839                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   

       Rating   Reviews                Size     Installs  Type Price  \
37        4.1   36639.0  Varies with device   5,000,000+  Free     0   
42        4.4  295221.0  Varies with device  10,000,000+  Free     0   
52        4.6   17057.0  Varies with device   1,000,000+  Free     0   
67        4.3   40211.0  Varies with device   5,000,000+  Free     0   
68        3.9     356.0  Varies with device     100,000+  Free     0   
...       ...       ...                 ...          ...   ...   ...   
10713     4.4    3471.0  Varies with device     100,000+  Free     0   
10725     3.6       8.0  Varies with device       1,000+  Free     0   
10765     3.4     622.0  Varies with device     100,000+  Free     0   
10826     4.0   88486.0  Varies with device   5,000,000+  Free     0   
10839     4.5     114.0  Varies with device       1,000+  Free     0   

      Content Rating             Genres        Last Updated  \
37          Everyone       Art & Design       July 14, 2018   
42          Everyone       Art & Design       July 30, 2018   
52          Everyone    Auto & Vehicles       July 30, 2018   
67          Everyone    Auto & Vehicles       July 30, 2018   
68          Everyone    Auto & Vehicles        May 25, 2018   
...              ...                ...                 ...   
10713       Everyone            Weather       July 24, 2018   
10725       Everyone  Maps & Navigation  September 27, 2017   
10765     Mature 17+             Social        May 23, 2018   
10826     Mature 17+             Social      March 23, 2018   
10839     Mature 17+  Books & Reference    January 19, 2015   

              Current Ver         Android Ver  
37     Varies with device        2.3.3 and up  
42     Varies with device  Varies with device  
52     Varies with device  Varies with device  
67     Varies with device  Varies with device  
68     Varies with device  Varies with device  
...                   ...                 ...  
10713  Varies with device  Varies with device  
10725  Varies with device          4.4 and up  
10765  Varies with device  Varies with device  
10826  Varies with device  Varies with device  
10839  Varies with device  Varies with device  

[1695 rows x 13 columns]

In [70]:
gp_store["Size"].unique()

array(['19M', '14M', '8.7M', '25M', '2.8M', '5.6M', '29M', '33M', '3.1M',
       '28M', '12M', '20M', '21M', '37M', '2.7M', '5.5M', '17M', '39M',
       '31M', '4.2M', '7.0M', '23M', '6.0M', '6.1M', '4.6M', '9.2M',
       '5.2M', '11M', '24M', 'Varies with device', '9.4M', '15M', '10M',
       '1.2M', '26M', '8.0M', '7.9M', '56M', '57M', '35M', '54M', '201k',
       '3.6M', '5.7M', '8.6M', '2.4M', '27M', '2.5M', '16M', '3.4M',
       '8.9M', '3.9M', '2.9M', '38M', '32M', '5.4M', '18M', '1.1M',
       '2.2M', '4.5M', '9.8M', '52M', '9.0M', '6.7M', '30M', '2.6M',
       '7.1M', '3.7M', '22M', '7.4M', '6.4M', '3.2M', '8.2M', '9.9M',
       '4.9M', '9.5M', '5.0M', '5.9M', '13M', '73M', '6.8M', '3.5M',
       '4.0M', '2.3M', '7.2M', '2.1M', '42M', '7.3M', '9.1M', '55M',
       '23k', '6.5M', '1.5M', '7.5M', '51M', '41M', '48M', '8.5M', '46M',
       '8.3M', '4.3M', '4.7M', '3.3M', '40M', '7.8M', '8.8M', '6.6M',
       '5.1M', '61M', '66M', '79k', '8.4M', '118k', '44M', '695k', '1.6M',
     

In [71]:
gp_store['Size'] = gp_store.Size.replace("Varies with device",np.nan)
gp_store['Size'] = gp_store.Size.str.replace("M","000") # All size values became the kilobyte type.
gp_store['Size'] = gp_store.Size.str.replace("k","")
gp_store['Size'] = gp_store.Size.replace("1,000+",1000)
gp_store['Size'] =gp_store['Size'].astype(float)
gp_store['Size'].dtype

dtype('float64')

In [72]:
gp_store['Size'].isnull().sum()

1695

In [73]:
gp_store["Size"].replace(np.nan, gp_store["Size"].mean(),inplace = True)

In [74]:
gp_store["Size"].unique()

array([1.90000000e+04, 1.40000000e+04, 8.70000000e+00, 2.50000000e+04,
       2.80000000e+00, 5.60000000e+00, 2.90000000e+04, 3.30000000e+04,
       3.10000000e+00, 2.80000000e+04, 1.20000000e+04, 2.00000000e+04,
       2.10000000e+04, 3.70000000e+04, 2.70000000e+00, 5.50000000e+00,
       1.70000000e+04, 3.90000000e+04, 3.10000000e+04, 4.20000000e+00,
       7.00000000e+00, 2.30000000e+04, 6.00000000e+00, 6.10000000e+00,
       4.60000000e+00, 9.20000000e+00, 5.20000000e+00, 1.10000000e+04,
       2.40000000e+04, 1.95772795e+04, 9.40000000e+00, 1.50000000e+04,
       1.00000000e+04, 1.20000000e+00, 2.60000000e+04, 8.00000000e+00,
       7.90000000e+00, 5.60000000e+04, 5.70000000e+04, 3.50000000e+04,
       5.40000000e+04, 2.01000000e+02, 3.60000000e+00, 5.70000000e+00,
       8.60000000e+00, 2.40000000e+00, 2.70000000e+04, 2.50000000e+00,
       1.60000000e+04, 3.40000000e+00, 8.90000000e+00, 3.90000000e+00,
       2.90000000e+00, 3.80000000e+04, 3.20000000e+04, 5.40000000e+00,
      

In [75]:
gp_store['Size'].isnull().sum()

0

### Installs

In [76]:
gp_store["Installs"].isnull().sum()

0

#### Install column has no null values 

In [77]:
gp_store["Installs"].unique()

array(['10,000+', '500,000+', '5,000,000+', '50,000,000+', '100,000+',
       '50,000+', '1,000,000+', '10,000,000+', '5,000+', '100,000,000+',
       '1,000,000,000+', '1,000+', '500,000,000+', '50+', '100+', '500+',
       '10+', '1+', '5+', '0+', '0'], dtype=object)

In [78]:
gp_store[gp_store["Installs"]=="Free"]

Empty DataFrame
Columns: [App, Category, Rating, Reviews, Size, Installs, Type, Price, Content Rating, Genres, Last Updated, Current Ver, Android Ver]
Index: []

In [79]:
gp_store["Installs"].unique()

array(['10,000+', '500,000+', '5,000,000+', '50,000,000+', '100,000+',
       '50,000+', '1,000,000+', '10,000,000+', '5,000+', '100,000,000+',
       '1,000,000,000+', '1,000+', '500,000,000+', '50+', '100+', '500+',
       '10+', '1+', '5+', '0+', '0'], dtype=object)

In [80]:
gp_store['Installs'] = gp_store.Installs.str.replace(",","")
gp_store['Installs'] = gp_store.Installs.str.replace("+","")
gp_store['Installs'] =gp_store['Installs'].astype(float)
gp_store['Installs'].dtype

<ipython-input-80-5468166d107d>:2: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  gp_store['Installs'] = gp_store.Installs.str.replace("+","")


dtype('float64')

In [81]:
gp_store["Installs"].unique()

array([1.e+04, 5.e+05, 5.e+06, 5.e+07, 1.e+05, 5.e+04, 1.e+06, 1.e+07,
       5.e+03, 1.e+08, 1.e+09, 1.e+03, 5.e+08, 5.e+01, 1.e+02, 5.e+02,
       1.e+01, 1.e+00, 5.e+00, 0.e+00])

In [82]:
gp_store[["Installs", "App"]].value_counts()

Installs     App                                              
100000000.0  ROBLOX                                               9
5000000.0    CBS Sports App - Scores, News, Stats & Watch Live    8
100000000.0  8 Ball Pool                                          7
10000000.0   ESPN                                                 7
500000000.0  Candy Crush Saga                                     7
                                                                 ..
10000.0      Android-P Mono Grey EMUI 8/5 Theme                   1
             Anime Expo 2018                                      1
             Anonymous caller detection                           1
             AppClose - The #1 FREE Co-Parenting App              1
100000.0     Blood Glucose Tracker                                1
Length: 9675, dtype: int64

### Type

In [83]:
gp_store["Type"].unique()

array(['Free', 'Paid', nan], dtype=object)

In [84]:
gp_store["Type"].isnull().sum()

1

#### type has one null value and it's converted or replace with other

In [85]:
gp_store["Type"].replace(np.nan,"Other",inplace = True)

In [86]:
gp_store["Type"].unique()

array(['Free', 'Paid', 'Other'], dtype=object)

In [87]:
gp_store["Type"].value_counts()

Free     10040
Paid       800
Other        1
Name: Type, dtype: int64

In [88]:
gp_store["Type"].isnull().sum()

0

### Price

In [89]:
gp_store["Price"].unique()

array(['0', '$4.99', '$3.99', '$6.99', '$1.49', '$2.99', '$7.99', '$5.99',
       '$3.49', '$1.99', '$9.99', '$7.49', '$0.99', '$9.00', '$5.49',
       '$10.00', '$24.99', '$11.99', '$79.99', '$16.99', '$14.99',
       '$1.00', '$29.99', '$12.99', '$2.49', '$10.99', '$1.50', '$19.99',
       '$15.99', '$33.99', '$74.99', '$39.99', '$3.95', '$4.49', '$1.70',
       '$8.99', '$2.00', '$3.88', '$25.99', '$399.99', '$17.99',
       '$400.00', '$3.02', '$1.76', '$4.84', '$4.77', '$1.61', '$2.50',
       '$1.59', '$6.49', '$1.29', '$5.00', '$13.99', '$299.99', '$379.99',
       '$37.99', '$18.99', '$389.99', '$19.90', '$8.49', '$1.75',
       '$14.00', '$4.85', '$46.99', '$109.99', '$154.99', '$3.08',
       '$2.59', '$4.80', '$1.96', '$19.40', '$3.90', '$4.59', '$15.46',
       '$3.04', '$4.29', '$2.60', '$3.28', '$4.60', '$28.99', '$2.95',
       '$2.90', '$1.97', '$200.00', '$89.99', '$2.56', '$30.99', '$3.61',
       '$394.99', '$1.26', '$1.20', '$1.04'], dtype=object)

In [90]:
gp_store["Price"].isnull().sum()

0

In [91]:
gp_store['Price'] = gp_store.Price.replace("Everyone",np.nan)

In [92]:
gp_store["Price"] = gp_store.Price.str.replace("$","").astype(float)

<ipython-input-92-f843bbae049d>:1: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  gp_store["Price"] = gp_store.Price.str.replace("$","").astype(float)


In [93]:
gp_store["Price"].unique()

array([  0.  ,   4.99,   3.99,   6.99,   1.49,   2.99,   7.99,   5.99,
         3.49,   1.99,   9.99,   7.49,   0.99,   9.  ,   5.49,  10.  ,
        24.99,  11.99,  79.99,  16.99,  14.99,   1.  ,  29.99,  12.99,
         2.49,  10.99,   1.5 ,  19.99,  15.99,  33.99,  74.99,  39.99,
         3.95,   4.49,   1.7 ,   8.99,   2.  ,   3.88,  25.99, 399.99,
        17.99, 400.  ,   3.02,   1.76,   4.84,   4.77,   1.61,   2.5 ,
         1.59,   6.49,   1.29,   5.  ,  13.99, 299.99, 379.99,  37.99,
        18.99, 389.99,  19.9 ,   8.49,   1.75,  14.  ,   4.85,  46.99,
       109.99, 154.99,   3.08,   2.59,   4.8 ,   1.96,  19.4 ,   3.9 ,
         4.59,  15.46,   3.04,   4.29,   2.6 ,   3.28,   4.6 ,  28.99,
         2.95,   2.9 ,   1.97, 200.  ,  89.99,   2.56,  30.99,   3.61,
       394.99,   1.26,   1.2 ,   1.04])

In [94]:
gp_store.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

In [95]:
gp_store.rename(columns = {"Price" : "Price ($)"},inplace = True)

In [96]:
gp_store.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price ($)', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

### Content Rating

In [97]:
gp_store["Content Rating"].unique()

array(['Everyone', 'Teen', 'Everyone 10+', 'Mature 17+',
       'Adults only 18+', 'Unrated'], dtype=object)

In [98]:
gp_store["Content Rating"].nunique()

6

In [99]:
gp_store["Content Rating"].isnull().sum()

0

In [100]:
gp_store[gp_store["Content Rating"] == "Unrated"]

App Category    Rating  Reviews  Size  Installs  \
7312     Best CG Photography   FAMILY  4.191513      1.0   2.5     500.0   
8266  DC Universe Online Map    TOOLS  4.100000   1186.0   6.4   50000.0   

      Type  Price ($) Content Rating         Genres       Last Updated  \
7312  Free        0.0        Unrated  Entertainment      June 24, 2015   
8266  Free        0.0        Unrated          Tools  February 27, 2012   

     Current Ver   Android Ver  
7312         5.2    3.0 and up  
8266         1.3  2.3.3 and up

### Genres

In [101]:
gp_store["Genres"].unique()

array(['Art & Design', 'Art & Design;Pretend Play',
       'Art & Design;Creativity', 'Art & Design;Action & Adventure',
       'Auto & Vehicles', 'Beauty', 'Books & Reference', 'Business',
       'Comics', 'Comics;Creativity', 'Communication', 'Dating',
       'Education;Education', 'Education', 'Education;Creativity',
       'Education;Music & Video', 'Education;Action & Adventure',
       'Education;Pretend Play', 'Education;Brain Games', 'Entertainment',
       'Entertainment;Music & Video', 'Entertainment;Brain Games',
       'Entertainment;Creativity', 'Events', 'Finance', 'Food & Drink',
       'Health & Fitness', 'House & Home', 'Libraries & Demo',
       'Lifestyle', 'Lifestyle;Pretend Play',
       'Adventure;Action & Adventure', 'Arcade', 'Casual', 'Card',
       'Casual;Pretend Play', 'Action', 'Strategy', 'Puzzle', 'Sports',
       'Music', 'Word', 'Racing', 'Casual;Creativity',
       'Casual;Action & Adventure', 'Simulation', 'Adventure', 'Board',
       'Trivia', 'Role 

In [102]:
gp_store["Genres"].nunique()

119

In [103]:
gp_store["Genres"].isnull().sum()

1

In [104]:
gp_store[gp_store["Genres"] == "nan"]

Empty DataFrame
Columns: [App, Category, Rating, Reviews, Size, Installs, Type, Price ($), Content Rating, Genres, Last Updated, Current Ver, Android Ver]
Index: []

In [105]:
gp_store["Genres"].replace(np.nan,"Others",inplace = True)

In [106]:
gp_store["Genres"].isnull().sum()

0

In [107]:
gp_store["Genres"].value_counts()

Tools                     842
Entertainment             623
Education                 549
Medical                   463
Business                  460
                         ... 
Arcade;Pretend Play         1
Card;Brain Games            1
Lifestyle;Pretend Play      1
Comics;Creativity           1
Strategy;Creativity         1
Name: Genres, Length: 120, dtype: int64

### Last Updated

In [108]:
gp_store["Last Updated"].unique()

array(['January 7, 2018', 'January 15, 2018', 'August 1, 2018', ...,
       'January 20, 2014', 'February 16, 2014', 'March 23, 2014'],
      dtype=object)

In [109]:
gp_store["Last Updated"].nunique()

1377

In [110]:
gp_store["Last Updated"].isnull().sum()

0

In [111]:
gp_store["Last Updated"].value_counts()

August 3, 2018       326
August 2, 2018       304
July 31, 2018        294
August 1, 2018       285
July 30, 2018        211
                    ... 
March 8, 2015          1
March 25, 2015         1
December 17, 2013      1
October 11, 2014       1
March 23, 2014         1
Name: Last Updated, Length: 1377, dtype: int64

In [112]:
gp_store["Last Updated"] = pd.to_datetime(gp_store["Last Updated"])

In [113]:
gp_store["Last Updated"]

0       2018-01-07
1       2018-01-15
2       2018-08-01
3       2018-06-08
4       2018-06-20
           ...    
10836   2017-07-25
10837   2018-07-06
10838   2017-01-20
10839   2015-01-19
10840   2018-07-25
Name: Last Updated, Length: 10841, dtype: datetime64[ns]

### Current Ver

In [114]:
gp_store["Current Ver"].unique()

array(['1.0.0', '2.0.0', '1.2.4', ..., '1.0.612928', '0.3.4', '2.0.148.0'],
      dtype=object)

In [115]:
gp_store["Current Ver"].nunique()

2831

In [116]:
gp_store["Current Ver"].isnull().sum()

8

In [117]:
gp_store[gp_store["Current Ver"] == "Varies with device"]

App             Category  \
3                              Sketch - Draw & Paint       ART_AND_DESIGN   
37                                Floor Plan Creator       ART_AND_DESIGN   
42                        Textgram - write on photos       ART_AND_DESIGN   
52                     Used Cars and Trucks for Sale    AUTO_AND_VEHICLES   
67                                Ulysse Speedometer    AUTO_AND_VEHICLES   
...                                              ...                  ...   
10725                                      Posta App  MAPS_AND_NAVIGATION   
10765                Chat For Strangers - Video Chat               SOCIAL   
10826      Frim: get new friends on local chat rooms               SOCIAL   
10839                  The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840  iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating   Reviews          Size    Installs  Type  Price ($)  \
3         4.5  215644.0  25000.000000  50000000.0  Free        0.0   
37        4.1   36639.0  19577.279477   5000000.0  Free        0.0   
42        4.4  295221.0  19577.279477  10000000.0  Free        0.0   
52        4.6   17057.0  19577.279477   1000000.0  Free        0.0   
67        4.3   40211.0  19577.279477   5000000.0  Free        0.0   
...       ...       ...           ...         ...   ...        ...   
10725     3.6       8.0  19577.279477      1000.0  Free        0.0   
10765     3.4     622.0  19577.279477    100000.0  Free        0.0   
10826     4.0   88486.0  19577.279477   5000000.0  Free        0.0   
10839     4.5     114.0  19577.279477      1000.0  Free        0.0   
10840     4.5  398307.0  19000.000000  10000000.0  Free        0.0   

      Content Rating             Genres Last Updated         Current Ver  \
3               Teen       Art & Design   2018-06-08  Varies with device   
37          Everyone       Art & Design   2018-07-14  Varies with device   
42          Everyone       Art & Design   2018-07-30  Varies with device   
52          Everyone    Auto & Vehicles   2018-07-30  Varies with device   
67          Everyone    Auto & Vehicles   2018-07-30  Varies with device   
...              ...                ...          ...                 ...   
10725       Everyone  Maps & Navigation   2017-09-27  Varies with device   
10765     Mature 17+             Social   2018-05-23  Varies with device   
10826     Mature 17+             Social   2018-03-23  Varies with device   
10839     Mature 17+  Books & Reference   2015-01-19  Varies with device   
10840       Everyone          Lifestyle   2018-07-25  Varies with device   

              Android Ver  
3              4.2 and up  
37           2.3.3 and up  
42     Varies with device  
52     Varies with device  
67     Varies with device  
...                   ...  
10725          4.4 and up  
10765  Varies with device  
10826  Varies with device  
10839  Varies with device  
10840  Varies with device  

[1459 rows x 13 columns]

In [118]:
gp_store[gp_store["Current Ver"] == ""]

Empty DataFrame
Columns: [App, Category, Rating, Reviews, Size, Installs, Type, Price ($), Content Rating, Genres, Last Updated, Current Ver, Android Ver]
Index: []

### Android Ver

In [119]:
gp_store["Android Ver"].unique()

array(['4.0.3 and up', '4.2 and up', '4.4 and up', '2.3 and up',
       '3.0 and up', '4.1 and up', '4.0 and up', '2.3.3 and up',
       'Varies with device', '2.2 and up', '5.0 and up', '6.0 and up',
       '1.6 and up', '1.5 and up', '2.1 and up', '7.0 and up',
       '5.1 and up', '4.3 and up', '4.0.3 - 7.1.1', '2.0 and up',
       '3.2 and up', '4.4W and up', '7.1 and up', '7.0 - 7.1.1',
       '8.0 and up', '5.0 - 8.0', '3.1 and up', '2.0.1 and up',
       '4.1 - 7.1.1', nan, '5.0 - 6.0', '1.0 and up', '2.2 - 7.1.1',
       '5.0 - 7.1.1'], dtype=object)

In [120]:
gp_store["Android Ver"].nunique()

33

In [121]:
gp_store["Android Ver"].isnull().sum()

2

In [122]:
gp_store[gp_store["Android Ver"] == "nan"]

Empty DataFrame
Columns: [App, Category, Rating, Reviews, Size, Installs, Type, Price ($), Content Rating, Genres, Last Updated, Current Ver, Android Ver]
Index: []

In [123]:
gp_store[gp_store["Android Ver"] == "Varies with device"]

App             Category  \
42                            Textgram - write on photos       ART_AND_DESIGN   
52                         Used Cars and Trucks for Sale    AUTO_AND_VEHICLES   
67                                    Ulysse Speedometer    AUTO_AND_VEHICLES   
68                                                REPUVE    AUTO_AND_VEHICLES   
85     CarMax – Cars for Sale: Search Used Car Inventory    AUTO_AND_VEHICLES   
...                                                  ...                  ...   
10713  My Earthquake Alerts - US & Worldwide Earthquakes              WEATHER   
10765                    Chat For Strangers - Video Chat               SOCIAL   
10826          Frim: get new friends on local chat rooms               SOCIAL   
10839                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating   Reviews          Size    Installs  Type  Price ($)  \
42        4.4  295221.0  19577.279477  10000000.0  Free        0.0   
52        4.6   17057.0  19577.279477   1000000.0  Free        0.0   
67        4.3   40211.0  19577.279477   5000000.0  Free        0.0   
68        3.9     356.0  19577.279477    100000.0  Free        0.0   
85        4.4   21777.0  19577.279477   1000000.0  Free        0.0   
...       ...       ...           ...         ...   ...        ...   
10713     4.4    3471.0  19577.279477    100000.0  Free        0.0   
10765     3.4     622.0  19577.279477    100000.0  Free        0.0   
10826     4.0   88486.0  19577.279477   5000000.0  Free        0.0   
10839     4.5     114.0  19577.279477      1000.0  Free        0.0   
10840     4.5  398307.0  19000.000000  10000000.0  Free        0.0   

      Content Rating             Genres Last Updated         Current Ver  \
42          Everyone       Art & Design   2018-07-30  Varies with device   
52          Everyone    Auto & Vehicles   2018-07-30  Varies with device   
67          Everyone    Auto & Vehicles   2018-07-30  Varies with device   
68          Everyone    Auto & Vehicles   2018-05-25  Varies with device   
85          Everyone    Auto & Vehicles   2018-08-04  Varies with device   
...              ...                ...          ...                 ...   
10713       Everyone            Weather   2018-07-24  Varies with device   
10765     Mature 17+             Social   2018-05-23  Varies with device   
10826     Mature 17+             Social   2018-03-23  Varies with device   
10839     Mature 17+  Books & Reference   2015-01-19  Varies with device   
10840       Everyone          Lifestyle   2018-07-25  Varies with device   

              Android Ver  
42     Varies with device  
52     Varies with device  
67     Varies with device  
68     Varies with device  
85     Varies with device  
...                   ...  
10713  Varies with device  
10765  Varies with device  
10826  Varies with device  
10839  Varies with device  
10840  Varies with device  

[1362 rows x 13 columns]

#### We are having some of the unwanted columns which will be of not much use in the analysis process. So let’s drop those columns.

In [124]:
gp_store.drop(["Current Ver","Android Ver","Last Updated"], axis = 1, inplace = True)

In [125]:
gp_store.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price ($)', 'Content Rating', 'Genres'],
      dtype='object')